packages/beacon-node/src/network/core/networkCore.ts

```diff
@@ -350,6 +359,15 @@ export class NetworkCore implements INetworkCore {
     return recipients.length;
   }
 
+  async setTargetGroupCount(count: number): Promise<void> {
+    this.networkConfig.setTargetGroupCount(count);
+  }
+
+  async setAdvertisedGroupCount(count: number): Promise<void> {
+    this.networkConfig.setAdvertisedGroupCount(count);
+    this.metadata.cgc = count;
```

Comment:
Not sure if there's a better way to handle syncing the metadata's CGC to the advertised group count -- maybe passing the metadata controller into the network config.

Reply:
I think this is probably ok.  It gets updated with the config so feels like we can be confident that the value is sync'd correctly